File: TP 3/listar.c
#include "listar.h"

int listar_superheroes(FILE* superheroes){

    heroe_t heroe;
    
    int leido = leer_superheroe(superheroes, &heroe);

    if(leido == EOF){
        printf("El archivo esta vacío no hay nada para mostrar\n");
        return ERROR_LISTAR;
    }

    char estado_completo[MAX_ESTADO];

    printf("------------------------------------------\n");
    while(leido != EOF){
     
        if(heroe.estado == VIVO){
            strcpy(estado_completo, VIVO_COMPLETO);
        }else{
            strcpy(estado_completo, MUERTO_COMPLETO);
        }

        printf("ID: %i - Nombre: %s - Edad: %i - Estado: %s\n", heroe.id, heroe.nombre, heroe.edad, estado_completo);
        printf("------------------------------------------\n");

        leido = leer_superheroe(superheroes, &heroe);
    }
    
    return NO_HAY_ERROR;
}

File: TP3 - PYTHON/listar.py
import heroe as hero
import codigo_error as code

#PRE: -
#POS: Muestra por pantalla la lista de superheroes. Retorna menos -2 si hubo un error. Sino 0.
def listar_superheroes(superheroes):
    
    heroe = hero.Heroes(0, "", 0, "")

    csvreader = hero.leer_superheroe(superheroes)

    estado_completo = " "
    
    print("------------------------------------------")
    for row in csvreader:
        datos = row.strip().split(";")
        heroe = hero.Heroes(datos[0], datos[1], datos[2], datos[3])
        if(heroe.estado == hero.VIVO):
            estado_completo = hero.VIVO_COMPLETO
        else:
            estado_completo = hero.MUERTO_COMPLETO
        
        print("ID:",heroe.id, "Nombre:",heroe.nombre, "Edad:",heroe.edad, "Estado:",estado_completo)
        print("------------------------------------------")

    return code.NO_HAY_ERROR
